### DISCUSSION TITLE
Problem starting pwrstatd

### DISCUSSION THREAD

**Discussion by VolvoBirkenstock** [mocaccinoOS/mocaccino#112] [General] [answered]
Hello, Mocaccino Gurus. 
I have a CyberPower UPS which should be monitored through USB with one of my computers.  This is done using the pwrstatd daemon.  I have used an install script to place the appropriate daemon and executables.  I have used systemctl while attempting to start the daemon.  (This method worked for Sabayon.)  
`$ systemctl enable pwrstatd.service`
This executed without a problem. 
`$ systemctl daemon-reload`
This executed without a problem. 
`$ systemctl restart pwrstatd.service` 
This was met with failure.  I see the following response after attempting to start the daemon: 
`Job for pwrstatd.service failed because the control process exited with error code.`
`See "systemctl status pwrstatd.service" and "journalctl -xeu pwrstatd.service" for details.`

`$ systemctl status pwrstatd.service`
`× pwrstatd.service - The monitor UPS software.`
`     Loaded: loaded (/etc/systemd/system/pwrstatd.service; enabled; preset: disabled)`
`     Active: failed (Result: exit-code) since Sat 2022-11-12 00:43:49 CST; 1min 31s ago`
`    Process: 14551 ExecStart=/etc/init.d/pwrstatd start (code=exited, status=1/FAILURE)`

`Nov 12 00:43:49 xXx systemd[1]: Starting The monitor UPS software....`
`Nov 12 00:43:49 xXx pwrstatd[14551]: /etc/init.d/pwrstatd: line 21: /lib/lsb/init-functions: No such file or directory`
`Nov 12 00:43:49 xXx systemd[1]: pwrstatd.service: Control process exited, code=exited, status=1/FAILURE`
`Nov 12 00:43:49 xXx systemd[1]: pwrstatd.service: Failed with result 'exit-code'.`
`Nov 12 00:43:49 xXx systemd[1]: Failed to start The monitor UPS software..`

`$ journalctl -xeu pwrstatd.service`
`Nov 12 00:43:49 xXx pwrstatd[14551]: /etc/init.d/pwrstatd: line 21: /lib/lsb/init-functions: No such file or directory`
`Nov 12 00:43:49 xXx systemd[1]: pwrstatd.service: Control process exited, code=exited, status=1/FAILURE`

I am apparently missing /lib/lsb/init-functions.  This worked with Sabayon.  How do I obtain this file? Did I miss something during the install process?  How can I fix this? 
Thank you very much in advance.

**Comment by msdobrescu** [comment]
Hi, I can't help you with this as I have no such UPS, but [here](https://unix.stackexchange.com/questions/9314/no-such-file-or-directory-etc-init-d-functions) is explained that reported missing scripts are distro specific. Gentoo seems not to provide those. Is there a chance you have used a Debian installation package? Have you try the .tar.gz one?
Alternatively, does nut work for you?

Extracting the .tar.gz and .deb, I see there are different install scripts. However, in .tar.gz the install script tries to match the distro. If none is matched, it peeks scripts that do not refer /lib/lsb/init-functions. You must ensure those are installed and used instead Debian, Redhat or Arch ones, namely default_pwrstatd. Then try again running it.

**Reply by VolvoBirkenstock** [reply]
Hello, @msdobrescu. 

The install script used for the CyberPower UPS is a tar.gz file supplied by CyberPower (version 1.4.1) and is the very same tar.gz file used for successful installation on Sabayon.  There is only one install shellscript named "install.sh".  Yes, the install shellscript does have internal conditional branching that supports multiple Linux operating systems including "suse", "red hat", "ubuntu", "debian", "raspian", "arch" and then a default.  It worked in the past with Sabayon without issue, so I don't understand why the installation would fail with Mocaccino unless there is a significant file difference between Sabayon and Mocaccino in the operating system.  I suppose it is possible that my Sabayon installation was larger than the Mocaccino distribution in terms of provided files.  No, I did not try "nut" and have no experience with it.  

I installed another operating system three days ago.  I have also installed Mocaccino within a VM and will continue to monitor Mocaccino on a frequent basis.  I very much hope to eventually return to using Mocaccino.  In my opinion, the Gentoo-based Sabayon/Mocaccino rolling update binary distribution model is convenient and provides the lowest operational latency.  I have noticed that the newly installed operating system is not as responsive as Sabayon/Mocaccino.  

Thank you very much for your assistance and diligence, msdobrescu.

**Comment by msdobrescu** [comment, marked as answer]
@VolvoBirkenstock, update the repos and try apps/powerpanel package.

**Reply by VolvoBirkenstock** [reply]
That works. :-)  
Again, thank you, @msdobrescu !!!

**Reply by msdobrescu** [reply]
Don't mention it!
